### DISCUSSION TITLE
Before archive but after stopping?

### DISCUSSION THREAD

**Discussion by modmew8** [offen/docker-volume-backup#498] [Q&A] [answered]
Hi there,

I just set up a backup for a docker compose application (consisting of a server and a postgres database) and wanted to make sure the server is stopped before the backup takes place. I thought of the following workflow:

1. stop the server using the `docker-volume-backup.stop-during-backup`-label
2. dump the database to a separate dump-volume using `docker-volume-backup.[step]-[pre|post]`
3. archive the server data-volume and the database dump-volume

After checking the logs, I assume `docker-volume-backup.archive-pre` takes place before the `archive` action, as well as before the `stop-during-backup` step. And `docker-volume-backup.archive-post` takes place after `archive` action, and also after the stopped containers have been restarted (so the dump is not within the backup).

So there is currently no possibility to execute a custom command after stopping the labeled containers but before archive step?

For now I settled with stopping and archiving the postgres data-volume (while postgres is still running, but with the server stopped I assume all open connections/transactions should have been terminated before the archive starts), but I wanted to ask about this usecase.

Best regards

**Comment by m90** [comment, marked as answer]
> So there is currently no possibility to execute a custom command after stopping the labeled containers but before archive step?

This is correct as of now, yes. I wonder whether it would make sense to add a `stop` lifecycle phase, but then this creates a nested ordering, which hasn't been there yet. Not sure.
